crates/flipperzero/src/furi/thread.rs

```diff
@@ -149,7 +154,7 @@ where
 pub fn current() -> Thread {
     use alloc::borrow::ToOwned;
 
-    let thread = unsafe { sys::furi_thread_get_current() };
+    let thread = unsafe { NonNull::new_unchecked(sys::furi_thread_get_current()) };
```

Comment:
This calls `pvTaskGetThreadLocalStoragePointer(NULL, 0)`. It does not check that the result is non-null, and there are other places inside the Flipper Zero firmware that call `pvTaskGetThreadLocalStoragePointer` and behave conditionally if it is null. However, the thread pointer is set via `vTaskSetThreadLocalStoragePointer` as the first action in `furi_thread_body` before calling `thread->callback`, and it should never be possible to call `thread::current()` in a thread outside of `thread->callback`, so this should be sound.